### PR TITLE
Feat: edit display name

### PR DESCRIPTION
**Pull request by pranita09**
### Describe your changes:

Edit Display Name functionality for the following pages.
- Database Service - DatabaseSchema - Table
- Dashboard Service
- Pipeline Service
- Messaging Service
- ML Model Service
- Container/Storage Service
- Search Service
- API Service

https://github.com/user-attachments/assets/09412890-e378-4b6d-9c41-b4f9218d6545

https://github.com/user-attachments/assets/3fe314b3-8960-454a-8576-122b0fb5b6a4

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx [hunk 1]
```diff
@@ -11,45 +11,69 @@
...
   const [schemas, setSchemas] = useState<DatabaseSchema[]>([]);
   const [isLoading, setIsLoading] = useState(true);
   const [showDeletedSchemas, setShowDeletedSchemas] = useState<boolean>(false);
 
+  const allowEditDisplayNamePermission = useMemo(() => {
+    return (
+      (!isVersionPage && permissions.databaseSchema.EditAll) ||
+      permissions.databaseSchema.EditDisplayName
+    );
+  }, [permissions, isVersionPage]);
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:75]: (!isVersionPage && (permissions.databaseSchema.EditAll ||
      permissions.databaseSchema.EditDisplayName ));

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx [hunk 2]
```diff
@@ -160,6 +184,95 @@ export const DatabaseSchemaTable = ({
...
+            schema.id === id
+              ? { ...schema, displayName: data.displayName }
+              : schema
+          )
+        );
+      } catch (error) {
+        showErrorToast(error as AxiosError);
+      }
+    },
+    [schemas, t]
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:208]: No need to add `t` import in the dependency.

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx [hunk 3]
```diff
@@ -160,6 +184,95 @@ export const DatabaseSchemaTable = ({
     }
   };
 
+  const handleDisplayNameUpdate = useCallback(
+    async (data: EntityName, id?: string) => {
+      try {
+        const schemaDetails = schemas.find((schema) => schema.id === id);
+        if (!schemaDetails) {
+          throw new Error(t('error.tableNotFound'));
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:192]: Can you re-check the translation file, I don't think we don't have any error JSON in the translation file.

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx [hunk 4]
```diff
@@ -69,6 +75,46 @@ function SchemaTablesTab({
...
+      (permissions.table.EditAll || permissions.table.EditDisplayName)
+    );
+  }, [permissions, isVersionView]);
+
+  const handleDisplayNameUpdate = useCallback(
+    async (data: EntityName, id?: string) => {
+      try {
+        const tableDetails = localTableData.find((table) => table.id === id);
+        if (!tableDetails) {
+          throw new Error(t('error.tableNotFound'));
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx:94]: same here around the error object in translation file.

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx [hunk 5]
```diff
@@ -69,6 +75,46 @@ function SchemaTablesTab({
...
+            table.id === id
+              ? { ...table, displayName: data.displayName }
+              : table
+          )
+        );
+      } catch (error) {
+        showErrorToast(error as AxiosError);
+      }
+    },
+    [localTableData, t]
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx:112]: we can remove this from here too

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx [hunk 6]
```diff
@@ -131,9 +149,96 @@ function ServiceMainTabContent({
     setIsEdit(false);
   };
 
+  const callServicePatchAPI = async (
+    serviceCategory: ServiceTypes,
+    id: string,
+    jsonPatch: any
+  ) => {
+    switch (serviceCategory) {
+      case 'databaseServices':
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:158]: We can use the `ServiceTypes` enum for the cases name instead of hardcoded one

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx [hunk 7]
```diff
@@ -131,9 +149,96 @@ function ServiceMainTabContent({
...
+        throw new Error('Unsupported service category');
+    }
+  };
+
+  const handleDisplayNameUpdate = useCallback(
+    async (entityData: EntityName, id?: string) => {
+      try {
+        const pageDataDetails = pageData.find((data) => data.id === id);
+        if (!pageDataDetails) {
+          throw new Error(t('error.databaseNotFound'));
```
Comment (Ashish8689) [inline openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:184]: same here

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+}));
+
+jest.mock('../../Modals/EntityNameModal/EntityNameModal.component', () => {
+  return {
+    __esModule: true,
+    default: jest.fn(() => (
+      <div data-testid="entity-name-modal">Mocked Modal</div>
+    )),
+  };
+});
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx:39]:
```suggestion
jest.mock(
'../../Modals/EntityNameModal/EntityNameModal.component',
  () => jest.fn().mockImplementation(() => <p>
    Mocked Modal</p>)
);
```

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+  it('Should render the component with name when display name is empty', async () => {
+    await act(async () => {
+      render(
+        <MemoryRouter>
+          <DisplayName {...mockProps} displayName={undefined} />
+        </MemoryRouter>
+      );
+
+      const nameField = await screen.getByTestId('column-name');
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx:80]: We don't need `await` for `getByTestId`.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+  const [isDisplayNameEditing, setIsDisplayNameEditing] = useState(false);
+
+  const handleDisplayNameUpdate = async (data: EntityName) => {
+    if (!onEditDisplayName) {
+      return;
+    }
+    setIsDisplayNameEditing(true);
+    try {
+      await onEditDisplayName(data, id);
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:52]:
```suggestion
    setIsDisplayNameEditing(true);
    try {
      await onEditDisplayName?.(data, id);
```

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+          data-testid="column-name">
+          {isEmpty(displayName) ? (
+            <Link className="break-word" data-testid={name} to={link}>
+              {name}
+            </Link>
+          ) : (
+            name
+          )}
+        </Typography.Text>
+      </div>
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:74]: Why do we need this outer div here?

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        <Typography.Text
+          className="m-b-0 d-block text-grey-muted break-word"
+          data-testid="column-name">
+          {isEmpty(displayName) ? (
+            <Link className="break-word" data-testid={name} to={link}>
+              {name}
+            </Link>
+          ) : (
+            name
+          )}
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:72]: Not sure why are we checking isEmpty on `displayName` here and showing `name` inside?

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+      {!isEmpty(displayName) ? (
+        // It will render displayName fallback to name
+        <Typography.Text
+          className="m-b-0 d-block break-word"
+          data-testid="column-display-name">
+          <Link className="break-word" data-testid={name} to={link}>
+            {displayName}
+          </Link>
+        </Typography.Text>
+      ) : null}
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:84]: Seems redundant to the above code. Can we merge it in a single condition?

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+        <Tooltip placement="right" title={t('label.edit')}>
+          <Button
+            className="cursor-pointer hover-cell-icon w-fit-content"
+            data-testid="edit-displayName-button"
+            style={{
+              color: DE_ACTIVE_COLOR,
+              padding: 0,
+              border: 'none',
+              background: 'transparent',
+            }}
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:96]: Let's not use style for this. We can make use of classNames for customised stylings

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+            onClick={() => setIsDisplayNameEditing(true)}>
+            <IconEdit style={{ color: DE_ACTIVE_COLOR, ...ICON_DIMENSION }} />
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:98]: Instead of passing as children, we can pass icon in the `icon` prop

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx [hunk 16]
```diff
@@ -69,6 +75,46 @@ function SchemaTablesTab({
   pagingInfo,
 }: Readonly<SchemaTablesTabProps>) {
   const { t } = useTranslation();
+  const [localTableData, setLocalTableData] = useState<Table[]>([]);
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx:78]: Let's not use a different state to store the same table data inside the child component. We can use the state from the parent itself.

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx [hunk 17]
```diff
@@ -131,9 +149,96 @@ function ServiceMainTabContent({
...
+        return await patchMlModelDetails(id, jsonPatch);
+      case 'storageServices':
+        return await patchContainerDetails(id, jsonPatch);
+      case 'searchServices':
+        return await patchSearchIndexDetails(id, jsonPatch);
+      case 'apiServices':
+        return await patchApiCollection(id, jsonPatch);
+      default:
+        throw new Error('Unsupported service category');
+    }
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:176]: Let's move this logic to the utils.

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx [hunk 18]
```diff
@@ -69,14 +83,18 @@ function ServiceMainTabContent({
...
   const { t } = useTranslation();
   const { serviceCategory } = useParams<{
     serviceCategory: ServiceTypes;
   }>();
 
   const { fqn: serviceFQN } = useFqn();
+  const { permissions } = usePermissionProvider();
+
   const [isEdit, setIsEdit] = useState(false);
+  const [pageData, setPageData] = useState<ServicePageData[]>([]);
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:97]: Same here, please avoid using a different local state to store the same data in the child.

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx (hunk 1 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:75]: Corrected the condition.

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx (hunk 2 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:208]: Removed `t` from dependency

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx (hunk 3 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:192]: Corrected it.

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx (hunk 4 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx:94]: Corrected it, removed `throw new Error` and used `return` instead.

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx (hunk 6 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:158]: Used `ServiceTypes` enum instead of strings.

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx (hunk 7 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:184]: Corrected it.

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx (hunk 17 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:176]: Moved this function to `ServiceMainTabContentUtils.tsx`

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx [hunk 19]
```diff
@@ -160,6 +184,95 @@ export const DatabaseSchemaTable = ({
...
+            </span>
+          ),
+      },
+      {
+        title: t('label.owner-plural'),
+        dataIndex: 'owners',
+        key: 'owners',
+        width: 120,
+        render: (owners: EntityReference[]) =>
+          !isUndefined(owners) && owners.length > 0 ? (
```
[the remark this answers is not in the document]
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:256]: Made the suggested changes.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx (hunk 8 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx:39]: Applied suggested changes.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx (hunk 9 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx:80]: Removed `await`.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx (hunk 10 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:52]: `onEditDisplayName` is optional prop. If in case it is undefined, it should be handled, so added the condition.

Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:52]: @pranita09, yes, that should be handled by my suggested code.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx (hunk 11 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:74]: Removed the outer div.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx (hunk 13 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:84]: Removed the redundant code, and used a single condition to show `name` and `displayName`

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx (hunk 12 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:72]: if `displayName` is empty then we need to add Link to the `name` otherwise `displayName`, that's why checking isEmpty on `displayName`

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx (hunk 15 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:98]: Used `icon` prop to pass the IconEdit

Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:98]: Used `icon` prop to pass the IconEdit

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx (hunk 14 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:96]: created class and used className for the customised styling

File: openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx (hunk 16 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/DatabaseSchemaPage/SchemaTablesTab.tsx:78]: Updated the code.

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx (hunk 18 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/ServiceDetailsPage/ServiceMainTabContent.tsx:97]: Updated the code.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 20]
```diff
@@ -59,44 +56,40 @@ const DisplayName: React.FC<DisplayNameProps> = ({
...
+            </Typography.Text>
+          </>
+        )}
+      </Typography.Text>
 
       {allowRename ? (
         <Tooltip placement="right" title={t('label.edit')}>
           <Button
-            className="cursor-pointer hover-cell-icon w-fit-content"
+            className="cursor-pointer hover-cell-icon w-fit-content display-name-edit-button"
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:83]: We can eliminate these styling classes and unify all stylings in the `display-name-edit-button` class

File: openmetadata-ui/src/main/resources/ui/src/utils/ServiceMainTabContentUtils.tsx [hunk 21]
```diff
@@ -124,3 +130,30 @@ export const getServiceMainTabColumns = (
...
+    case ServiceCategory.ML_MODEL_SERVICES:
+      return await patchMlModelDetails(id, jsonPatch);
+    case ServiceCategory.STORAGE_SERVICES:
+      return await patchContainerDetails(id, jsonPatch);
+    case ServiceCategory.SEARCH_SERVICES:
+      return await patchSearchIndexDetails(id, jsonPatch);
+    case ServiceCategory.API_SERVICES:
+      return await patchApiCollection(id, jsonPatch);
+    default:
+      throw new Error('Unsupported service category');
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/utils/ServiceMainTabContentUtils.tsx:157]: Let's not throw an error in case of no match. You can return a void promise.

File: openmetadata-ui/src/main/resources/ui/webpack.config.dev.js [hunk 22]
```diff
@@ -19,7 +19,8 @@ const process = require('process');
 
 const outputPath = path.join(__dirname, 'build');
 const subPath = process.env.APP_SUB_PATH ?? '';
-const devServerTarget = process.env.DEV_SERVER_TARGET ?? 'http://localhost:8585/';
+const devServerTarget =
+  process.env.DEV_SERVER_TARGET ?? 'https://sandbox-beta.open-metadata.org/';
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/webpack.config.dev.js:23]: Seems like it was committed by mistake. Can you please revert this?

Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/webpack.config.dev.js:23]: Reverted the changes.

File: openmetadata-ui/src/main/resources/ui/src/styles/app.less [hunk 23]
```diff
@@ -843,4 +843,11 @@ a[href].link-text-grey,
...
   border: none;
   background-color: transparent;
+  cursor: pointer;
+  opacity: 0;
+  transition: 0.3s ease-in;
+  width: fit-content;
+  &:hover {
+    opacity: 100;
+  }
 }
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/styles/app.less:853]: Can we move this styling to component specific less file? also the hover effect seems to be applied on the button itself, it should be on whole displayName component

Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/styles/app.less]: Removed these styles from app.less and fixed the hover effect, it is applied on the whole displayName component.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+            className="hover-cell-icon"
+            data-testid="edit-displayName-button"
+            icon={
+              <IconEdit style={{ color: DE_ACTIVE_COLOR, ...ICON_DIMENSION }} />
+            }
+            style={{
+              padding: '0',
+              border: 'none',
+              background: 'transparent',
+            }}
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:92]: Please avoid inline styling. It looks like you can achieve this by the `ghost` prop.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+        )}
+      </Typography.Text>
+
+      {allowRename ? (
+        <Tooltip placement="right" title={t('label.edit')}>
+          <Button
+            className="hover-cell-icon"
+            data-testid="edit-displayName-button"
+            icon={
+              <IconEdit style={{ color: DE_ACTIVE_COLOR, ...ICON_DIMENSION }} />
```
Comment (aniketkatkar97) [inline openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:86]: Same here, please provide styling by classNames. We already have classNames defined for common stylings

Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:86]: Removed inline styles.

File: openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx (hunk 24 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.tsx:92]: Removed inline styles

**Review by aniketkatkar97 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceVersionPage/ServiceVersionMainTabContent.tsx [hunk 26]
```diff
@@ -42,15 +43,24 @@ function ServiceVersionMainTabContent({
...
+  const editDisplayNamePermission = useMemo(() => {
+    return !isVersionPage
+      ? permissions.databaseService.EditAll ||
+          permissions.databaseService.EditDisplayName
+      : false;
+  }, [permissions]);
 
   const tableColumn: ColumnsType<ServicePageData> = useMemo(
-    () => getServiceMainTabColumns(serviceCategory),
+    () => getServiceMainTabColumns(serviceCategory, editDisplayNamePermission),
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/ServiceVersionPage/ServiceVersionMainTabContent.tsx:63]: its already version page, why we need to check permission here?

File: openmetadata-ui/src/main/resources/ui/webpack.config.dev.js [hunk 27]
```diff
@@ -19,7 +19,8 @@ const process = require('process');
 
 const outputPath = path.join(__dirname, 'build');
 const subPath = process.env.APP_SUB_PATH ?? '';
-const devServerTarget = process.env.DEV_SERVER_TARGET ?? 'http://localhost:8585/';
+const devServerTarget =
+  process.env.DEV_SERVER_TARGET ?? 'http://localhost:8585/';
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/webpack.config.dev.js:23]: revert this change

**Review by ShaileshParmar11 — COMMENTED** [review]
left few comment

File: openmetadata-ui/src/main/resources/ui/src/pages/ServiceVersionPage/ServiceVersionMainTabContent.tsx (hunk 26 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/pages/ServiceVersionPage/ServiceVersionMainTabContent.tsx:63]: Removed edit permission from version page

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx [hunk 28]
```diff
@@ -192,7 +192,9 @@ export const DatabaseSchemaTable = ({
         if (!schemaDetails) {
           return;
         }
-        const updatedData = { ...schemaDetails, displayName: data.displayName };
+        const updatedData = data.displayName
+          ? { ...schemaDetails, displayName: data.displayName }
+          : { ...schemaDetails, displayName: undefined };
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx:197]: We can improve here, Lets connect for that!

**Review by ShaileshParmar11 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx (hunk 28 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/src/components/Database/DatabaseSchema/DatabaseSchemaTable/DatabaseSchemaTable.tsx]: Refactored the code.

File: openmetadata-ui/src/main/resources/ui/webpack.config.dev.js (hunk 27 above)
Comment (pranita09) [reply openmetadata-ui/src/main/resources/ui/webpack.config.dev.js]: Reverted.

**Review by ShaileshParmar11 — APPROVED** [review]
